### ISSUE TITLE
Testing: ability to quickly reset the database for E2E testing

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#10965]
Currently E2E tests have to clean up after themselves. This is annoying and error prone. It would be nice if you could reset the database before a group of tests are run, e.x. using `beforeAll`:

```
import { resetDatabase } from "__support/e2e";

describe("whatever", () => {
  beforeAll(() => resetDatabase());

  it("...", () => {
    // test1 
  })

  it("...", () => {
    // test2
  })
})
```

Or if it's fast enough, even reset before every test is run using `beforeEach`:

```
import { resetDatabase } from "__support/e2e";

describe("whatever", () => {
  beforeEach(() => resetDatabase());

  it("...", () => {
    // test1 
  })

  it("...", () => {
    // test2
  })
})
```

`resetDatabase` would probably just make a HTTP request to an endpoint that's only exposed during E2E tests. It would do some magic to load a fresh copy of the fixture H2 database (`frontend/test/__runner__/test_db_fixture.db.mv.db`) without restarting the backend. The faster the better since it will probably be used at a minimum between each file's worth of E2E tests.

One thing we'll probably have to change is we don't currently update the E2E fixture database when we run migrations.

Also, this would be a good opportunity to make sure the backend is stateless.

**Comment by paulrosenzweig** [comment]
As we're about to port e2e tests over to Cypress, it'd be awesome to have this feature. Here are some hypothetical options for how the reset endpoint could work just to get the conversation started:

1. Add endpoints to begin/rollback a transaction
This sounds gross, but it might be the most performant.
2. Copy the H2 db on disk before loading it. On reset, we'd discard and load it again.
The e2e test runner already does this once before booting the server. The fixture is copied to /tmp, so that tests don't edit it. Here we'd want to do that within one boot of the serve.
3. Execute SQL against the db to reset it
If we defined the fixture in a text file (as SQL? JSON?), we could truncate each table and repopulate it.

@camsaul what are your thoughts?

**Comment by tlrobinson** [comment]
One orthogonal idea would be to define the fixture db(s) using our serialization mechanism, then when we start the E2E tests generate either a H2 database from it (for paul's option 2), or the SQL commands (for paul's option 3). That would be nicer than committing the binary H2 database like we currently do.

**Comment by kdoh** [comment]
I dig that. Feels like it'd be both a good way to use that mechanism internally and also it'd give us more transparency into what those datasets look like and make it easier to edit them. The H2 test databases are a bit of a black box unless you spin up MB itself against them.

**Comment by paulrosenzweig** [comment]
Closed in #11564
